static/js/public/snap-details/videos.js

```diff
@@ -1,29 +1,71 @@
-function youtube(holderEl, url) {
-  const frame = document.createElement("iframe");
-  frame.id = "ytplayer";
-  frame.type = "text/html";
-  frame.width = 440;
-  frame.height = 248;
-  frame.src = `${url}?autoplay=0&origin=${window.location.href}`;
-  frame.setAttribute("frameborder", "0");
-  holderEl.appendChild(frame);
+function vimeo() {
+  const vimeoPlayerScript = document.createElement("script");
+  vimeoPlayerScript.src = "https://player.vimeo.com/api/player.js";
+  const firstScript = document.getElementsByTagName("script")[0];
+  firstScript.parentNode.insertBefore(vimeoPlayerScript, firstScript);
+
+  const frame = document.getElementById("vimeoplayer");
+
+  const vimeoReady = () => {
+    const player = new window.Vimeo.Player(frame);
+    player.on("play", function() {
+      player.setVolume(0);
+    });
+    player.play();
+  };
+
+  const checkVimeo = () => {
+    if (window.Vimeo) {
+      vimeoReady();
+    } else {
+      setTimeout(checkVimeo, 200);
+    }
+  };
+
+  checkVimeo();
+}
+
+function asciinema(holderEl) {
+  const asciinemaPlayer = holderEl.querySelector("iframe");
+
+  if (!asciinemaPlayer) {
+    setTimeout(asciinema.bind(this, holderEl), 200);
+    return;
+  }
+
+  const parent = asciinemaPlayer.parentNode;
+  const blocker = holderEl.querySelector(".p-carousel__item-blocker");
+
+  blocker.addEventListener("click", e => {
+    e.stopImmediatePropagation();
+    if (document.fullscreenEnabled) {
+      parent.requestFullscreen();
+    }
+  });
+
+  document.addEventListener("fullscreenchange", () => {
```

Comment:
What is this script for? Is it possible to open asciinema full screen? I don't seem to see any controls for that.

Reply:
As asciinema "videos" can be any width and height (as recorded by the uploader) we're limiting the width/ height.

If the "video" is wider and taller than the container the control buttons are hidden.

We added clicking on the "video" to open a faux full-screen mode, because you can't access the asciinema iframe content to trigger a full-screen from their code.

Reply:
Ok, so asciinema controls are hidden (I don't see them), but I also don't know how to trigger this fullscreen mode. Clicking on the video doesn't do anything.